ql/src/java/org/apache/hadoop/hive/ql/security/authorization/command/CommandAuthorizerV2.java

```diff
@@ -185,6 +189,38 @@ private static void addHivePrivObject(Entity privObject, Map<String, List<String
           tableName2Cols.get(Table.getCompleteName(table.getDbName(), table.getTableName()));
       hivePrivObject = new HivePrivilegeObject(privObjType, table.getDbName(), table.getTableName(),
           null, columns, actionType, null, null, table.getOwner(), table.getOwnerType());
+      if(table.getStorageHandler() != null){
+        //TODO: add hive privilege object for storage based handlers for create and alter table commands.
+        if(hiveOpType == HiveOperationType.CREATETABLE ||
+                hiveOpType == HiveOperationType.ALTERTABLE_PROPERTIES ||
+                hiveOpType == HiveOperationType.CREATETABLE_AS_SELECT){
+          String storageuri = null;
+          Map<String, String> tableProperties = new HashMap<>();
+          Configuration conf = new Configuration();
+          tableProperties.putAll(table.getSd().getSerdeInfo().getParameters());
+          tableProperties.putAll(table.getParameters());
+          try {
+            Method methodIsImplemented = table.getStorageHandler().getClass().getMethod("getURIForAuth", Map.class);
+            if(methodIsImplemented != null && table.getStorageHandler() instanceof DefaultStorageHandler) {
+              DefaultStorageHandler defaultHandler = (DefaultStorageHandler) ReflectionUtils.newInstance(
+                      conf.getClassByName(table.getStorageHandler().getClass().getName()), SessionState.get().getConf());
+              storageuri = defaultHandler.getURIForAuth(tableProperties).toString();
+            }else if(methodIsImplemented != null && table.getStorageHandler() instanceof HiveStorageAuthorizationHandler){
+              HiveStorageAuthorizationHandler authorizationHandler = (HiveStorageAuthorizationHandler) ReflectionUtils.newInstance(
+                      conf.getClassByName(table.getStorageHandler().getClass().getName()), SessionState.get().getConf());
+              storageuri = authorizationHandler.getURIForAuth(tableProperties).toString();
+            }
+          }catch(NullPointerException nullExp){
+            throw nullExp;
+          }catch(Exception ex){
+            //Custom storage handler that has not implemented the getURIForAuth()
+            storageuri = table.getStorageHandler().getClass().getName()+"://"+
```

Comment:
if the above change makes sense, then this code should go into the else block instead of the exception handler.